File: lib_common/src/main/java/com/pfl/common/base/BaseActivity.java
package com.pfl.common.base;

import androidx.appcompat.app.AppCompatActivity;
import com.jojo.design.common_ui.dialog.LoadingDialog;

public class BaseActivity extends AppCompatActivity {

    private LoadingDialog mDialog;

    public void showDialog() {
        if (mDialog == null || !mDialog.isShowing()) {
            mDialog = new LoadingDialog(this);
            mDialog.show();
        }
    }

    public void dismissDialog() {
        if (null != mDialog && mDialog.isShowing()) {
            mDialog.dismiss();
        }
    }
}

File: lib_common/src/main/java/com/pfl/common/base/BaseActivity.kt
package com.pfl.common.base

import androidx.appcompat.app.AppCompatActivity
import com.jojo.design.common_ui.dialog.LoadingDialog

open class BaseActivity : AppCompatActivity() {

    private var mDialog: LoadingDialog? = null

    fun showDialog() {
        if (mDialog == null || !mDialog!!.isShowing()) {
            mDialog = LoadingDialog(this)
            mDialog!!.show()
        }
    }

    fun dismissDialog() {
        if (null != mDialog && mDialog!!.isShowing()) {
            mDialog!!.dismiss()
        }
    }
}
